File: data_management/Models/Equipment.cs
namespace DataManagement.Models;

public class Equipment
{
    public int EquipmentId { get; set; }
    public string EquipmentName { get; set; } = string.Empty;
    public string? Manufacturer { get; set; }
    public string? Model { get; set; }
    public string? SerialNumber { get; set; }
    public short? PurchaseYear { get; set; }
    public DateOnly? CalibrationDue { get; set; }
    public string? Location { get; set; }
    public string? ConnectingStr { get; set; }
    public string? Notes { get; set; }
    public ICollection<LabEquipment>? LabEquipments { get; set; }
    public ICollection<Measurement>? Measurements { get; set; }
}

File: frontend/src/types/equipment.ts
export type Equipment = {
    equipmentId: number,
    equipmentName: string,
    manufacturer: string | null,
    model: string | null,
    serialNumber: string | null,
    purchaseYear: number | null,
    calibrationDue: string | null,
    location: string | null,
    connectingStr: string | null,
    notes: string | null,
}

export type CreateEquipmentRequest = {
    equipmentName: string,
    manufacturer?: string,
    model?: string,
    serialNumber?: string,
    purchaseYear?: number,
    calibrationDue?: string,
    location?: string,
    connectingStr?: string,
    notes?: string,
}